src/client/serialize/daos_serialize.c

```diff
@@ -637,41 +637,49 @@ deserialize_roots(hid_t file_id, struct daos_prop_entry *entry, const char *prop
 	attr_dtype = H5Aget_type(cont_attr);
 	if (attr_dtype < 0) {
 		D_ERROR("failed to get attribute type\n");
-		D_GOTO(out, rc = -DER_MISC);
+		D_GOTO(out_attr, rc = -DER_MISC);
 	}
 	attr_dtype_size = H5Tget_size(attr_dtype);
 	if (attr_dtype_size < 0) {
 		D_ERROR("failed to get attribute type size\n");
-		D_GOTO(out, rc = -DER_MISC);
+		D_GOTO(out_dtype, rc = -DER_MISC);
```

Comment:
I've got a PR to remove this check in https://github.com/daos-stack/daos/pull/12311 H5Tget_size returns a signed value so the `<0` test is meaningless.

Reply:
It returns 0 on error. I have this coverity PR already for that
https://github.com/daos-stack/daos/pull/12317/files

Reply:
https://docs.hdfgroup.org/hdf5/develop/group___h5_t.html#ga1b971589cd7a86f3e84affdee455564e

Reply:
Though I see now `attr_dtype_size` isn't used anywhere. Since this is a coverity issue, I think it's better to let it be tracked by the coverity ticket: https://daosio.atlassian.net/browse/DAOS-13596

Reply:
I'll defer to your fix.  It's reassuring that coverity and the compiler found the same issue and an indication we should enable those extra checks.

Reply:
Thanks. I Merged #12317 into this one for simplicity, and also removed that check altogether since the size isn't needed.